providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py

```diff
@@ -291,8 +291,7 @@ def _process_workloads(self, workloads: list[workloads.All]) -> None:
             key = w.ti.key  # type: ignore[union-attr]
             queue = w.ti.queue  # type: ignore[union-attr]
 
-            # TODO: will be handled by https://github.com/apache/airflow/issues/46892
-            executor_config = {}  # type: ignore[var-annotated]
+            executor_config = w.ti.executor_config or {}  # type: ignore[union-attr]
```

Comment:
Instead of the "ignore[union-attr]" we should do a type check and fail if `w` isn't ExecuteTask. Drive by refactor :)

Reply:
Thank you, TIL that mypy can be smart enough to gauge that :)